svm/src/account_loader.rs

```diff
@@ -132,7 +133,7 @@ pub fn validate_fee_payer(
     payer_account: &mut AccountSharedData,
     payer_index: IndexOfAccount,
     error_metrics: &mut TransactionErrorMetrics,
-    rent_collector: &RentCollector,
+    rent_manager: Option<&dyn SVMRentManager>,
```

Comment:
It's optional at the SVM's entrypoint level (`TransactionProcessingEnvironment`), so I piped the optional value all the way through and then just handled the `None` case during rent collection.

Maybe this isn't the way to go, though? Previously we were doing `unwrap_or_default` on the optional `RentCollector`. Do you think it's better if I continue with that behavior, and retain the required parameter for this methods?

Reply:
I think we should keep the simplest method. Previously we would only handle the optional parameter once and then pass on a default value to all the calls. Now, every function must deal with the optional separately.

I believe dealing with optional only once is preferable.

Reply:
Yeah I think you're right. That means we'll have to include a `RentManager` struct within `solana-svm` that implements the trait and can be used when the user provides `None`. However, that might be misleading in thinking that you're disabling rent collection by providing `None`, when you'd really be defaulting to the protocol. This can probably be remedied with good documentation, though.

Reply:
Allowing developers to disable rent collection is a good idea. I think it needs a more thought. Currently, there is a feature to disable rent collection and accounts whose rent epoch is `u64::MAX` are rent-exempt.

Reply:
https://github.com/anza-xyz/agave/blob/7707ebde339ef6edad39adb323a50dc255f3240b/sdk/src/feature_set.rs#L724